src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseParticipationService.java

```diff
@@ -152,7 +152,9 @@ public boolean canAccessParticipation(@NotNull ProgrammingExerciseParticipation
         if (participation instanceof ProgrammingExerciseStudentParticipation studentParticipation && studentParticipation.isOwnedBy(user)) {
             return true;
         }
-        final var programmingExercise = programmingExerciseRepository.getProgrammingExerciseFromParticipation(participation);
+
+        ProgrammingExercise programmingExercise = participation.getProgrammingExercise() != null ? participation.getProgrammingExercise()
+                : programmingExerciseRepository.getProgrammingExerciseFromParticipation(participation);
```

Comment:
Isn’t this basically what `getProgrammingExerciseFromParticipation` does internally as well? Why is it needed now?

Reply:
Oops this somehow slipped through. I reverted that change